src/util/expr_cast.h

```diff
@@ -78,7 +79,7 @@ auto expr_try_dynamic_cast(TExpr &base)
   static_assert(
     std::is_base_of<exprt, T>::value,
     "The template argument T must be derived from exprt.");
-  if(!can_cast_expr<T>(base))
+  if(!can_cast_expr<typename std::decay<T>::type>(base))
```

Comment:
Fixes some issues casting to `const thingt` rather than just `thingt`

Reply:
I think you meant `std:remove_const`. This one removes `&`, `&&` and `volatile` if I'm not mistaken.